README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# oica <a href='https://jhelvy.github.io/oica/'><img src='man/figures/logo.png' align="right" style="height:139px;"/></a>

<!-- badges: start -->

[![CRAN
status](https://www.r-pkg.org/badges/version/oica)](https://CRAN.R-project.org/package=oica)
<!-- badges: end -->

This package contains some tidy formatted data on vehicle production and
sales from the [Organisation Internationale des Constructeurs
d’Automobiles (OICA)](https://www.oica.net/) (English Name:
“International Organization of Motor Vehicle Manufacturers”). The data
sets are also merged with standard country / area codes come from the
[United Nations Statistics
Division](https://unstats.un.org/unsd/methodology/m49/overview/). The
package contains the following data sets:

| Name            | Description                                                                                               |
|-----------------|-----------------------------------------------------------------------------------------------------------|
| `production`    | Vehicle [production statistics](https://www.oica.net/category/production-statistics/) by country and type |
| `sales_country` | Vehicle [sales statistics](https://www.oica.net/category/sales-statistics/) by country and type           |
| `sales_region`  | Vehicle [sales statistics](https://www.oica.net/category/sales-statistics/) by region and type            |

## Installation

The current version is not yet on CRAN, but you can install it from
Github using the {remotes} library:

``` r
# install.packages("remotes")
remotes::install_github("jhelvy/oica")
```

Load the library with:

``` r
library("oica")
```

## Usage

Once loaded, you can work with any of the three data sets. You can find
out more about each data set using `?`, e.g., `?production`.

Here is a quick preview of each data frame:

``` r
head(production)
#>   year   country type      n   region                       subregion
#> 1 2006 Argentina   pv 263120 Americas Latin America and the Caribbean
#> 2 2006 Argentina   cv 168981 Americas Latin America and the Caribbean
#> 3 2006 Australia   pv 270000  Oceania       Australia and New Zealand
#> 4 2006 Australia   cv  60900  Oceania       Australia and New Zealand
#> 5 2006   Austria   pv 248059   Europe                  Western Europe
#> 6 2006   Austria   cv  26873   Europe                  Western Europe
#>   intermediate_region least_developed land_locked_developing
#> 1       South America               0                      0
#> 2       South America               0                      0
#> 3                <NA>               0                      0
#> 4                <NA>               0                      0
#> 5                <NA>               0                      0
#> 6                <NA>               0                      0
#>   small_island_developing code_region code_subregion code_intermediate_region
#> 1                       0          19            419                        5
#> 2                       0          19            419                        5
#> 3                       0           9             53                       NA
#> 4                       0           9             53                       NA
#> 5                       0         150            155                       NA
#> 6                       0         150            155                       NA
#>   code_m49 code_iso_alpha2 code_iso_alpha3
#> 1       32              AR             ARG
#> 2       32              AR             ARG
#> 3       36              AU             AUS
#> 4       36              AU             AUS
#> 5       40              AT             AUT
#> 6       40              AT             AUT
head(sales_country)
#>   year country type    n region       subregion intermediate_region
#> 1 2005 Albania   pc  800 Europe Southern Europe                <NA>
#> 2 2006 Albania   pc  800 Europe Southern Europe                <NA>
#> 3 2007 Albania   pc 1600 Europe Southern Europe                <NA>
#> 4 2008 Albania   pc 1600 Europe Southern Europe                <NA>
#> 5 2009 Albania   pc 1000 Europe Southern Europe                <NA>
#> 6 2010 Albania   pc 1600 Europe Southern Europe                <NA>
#>   least_developed land_locked_developing small_island_developing code_region
#> 1               0                      0                       0         150
#> 2               0                      0                       0         150
#> 3               0                      0                       0         150
#> 4               0                      0                       0         150
#> 5               0                      0                       0         150
#> 6               0                      0                       0         150
#>   code_subregion code_intermediate_region code_m49 code_iso_alpha2
#> 1             39                       NA        8              AL
#> 2             39                       NA        8              AL
#> 3             39                       NA        8              AL
#> 4             39                       NA        8              AL
#> 5             39                       NA        8              AL
#> 6             39                       NA        8              AL
#>   code_iso_alpha3
#> 1             ALB
#> 2             ALB
#> 3             ALB
#> 4             ALB
#> 5             ALB
#> 6             ALB
head(sales_region)
#>   year region type      n
#> 1 2005 Africa   pc 784237
#> 2 2005 Africa   cv 328780
#> 3 2006 Africa   pc 926966
#> 4 2006 Africa   cv 387309
#> 5 2007 Africa   pc 939201
#> 6 2007 Africa   cv 382773
```

## Author, Version, and License Information

-   Author: *John Paul Helveston* <https://www.jhelvy.com/>
-   Date First Written: *Friday April 22, 2022*
-   License:
    [MIT](https://github.com/jhelvy/oica/blob/master/LICENSE.md)

## Citation Information

If you use this package for in a publication, I would greatly appreciate
it if you cited it - you can get the citation by typing
`citation("oica")` into R:

``` r
citation("oica")
#> 
#> To cite cbcTools in publications use:
#> 
#>   John Paul Helveston (2022). oica: Description.
#> 
#> A BibTeX entry for LaTeX users is
#> 
#>   @Manual{,
#>     title = {oica: Description},
#>     author = {John Paul Helveston},
#>     year = {2022},
#>     note = {R package version 0.0.1},
#>     url = {https://jhelvy.github.io/oica/},
#>   }
```


NEWS.md
# oica 0.0.1

- Added a `NEWS.md` file to track changes to the package.


R/data.R
#' Description
#'
#' Detailed description
#'
#' @format
#' Variable | Description
#' -------- | ---------------------------------------------
#' `var` | Description
#'
#' @docType data
#'
#' @usage data(production)
#'
#' @keywords datasets
#'
#' @source Raw data downloaded from \href{url}{text}
#'
#' @examples
#' data(production)
#'
#' head(production)
"production"




#' Description
#'
#' Detailed description
#'
#' @format
#' Variable | Description
#' -------- | ---------------------------------------------
#' `var` | Description
#'
#' @docType data
#'
#' @usage data(sales_country)
#'
#' @keywords datasets
#'
#' @source Raw data downloaded from \href{url}{text}
#'
#' @examples
#' data(sales_country)
#'
#' head(sales_country)
"sales_country"

#' Description
#'
#' Detailed description
#'
#' @format
#' Variable | Description
#' -------- | ---------------------------------------------
#' `var` | Description
#'
#' @docType data
#'
#' @usage data(sales_region)
#'
#' @keywords datasets
#'
#' @source Raw data downloaded from \href{url}{text}
#'
#' @examples
#' data(sales_region)
#'
#' head(sales_region)
"sales_region"


data-raw/format_sales.R
library(tidyverse)
library(readxl)
options(dplyr.width = Inf)

# 2005 - 2019
# Read in and merge the two data frames  
pc <- read_excel(file.path('salesData', 'raw', 'pc-sales-2019.xlsx'), skip = 5)
pc$country <- str_to_title(pc$'REGIONS/COUNTRIES')
pc <- pc %>%
    gather(year, sales, '2005':'2019') %>%
    select(country, year, sales) %>%
    mutate(type = 'pc')
cv <- read_excel(file.path('salesData', 'raw', 'cv-sales-2019.xlsx'), skip = 5)
cv$country <- str_to_title(cv$'REGIONS/COUNTRIES')
cv <- cv %>%
    gather(year, sales, '2005':'2019') %>%
    select(country, year, sales) %>%
    mutate(type = 'cv')
df19 <- rbind(pc, cv) %>%
    filter(is.na(country)==F) %>%
    mutate(
        country = str_replace(country, '\\*', ''),
        year = as.numeric(year)) %>%
    # Drop footnote rows
    filter(
        country != '1 Only Lv', 
        country != '2 Including Heavy Trucks, Buses And Coaches'
    )

# 2020 - 2021
pc <- read_excel(file.path('salesData', 'raw', 'pc-sales-2021.xlsx'), skip = 3)
pc$country <- str_to_title(pc$'REGIONS/COUNTRIES')
pc <- pc %>%
    gather(year, sales, 'Q1-Q4 2019':'Q1-Q4 2021') %>%
    select(country, year, sales) %>%
    mutate(type = 'pc')
cv <- read_excel(file.path('salesData', 'raw', 'cv-sales-2021.xlsx'), skip = 3)
cv$country <- str_to_title(cv$'REGIONS/COUNTRIES')
cv <- cv %>%
    gather(year, sales, 'Q1-Q4 2019':'Q1-Q4 2021') %>%
    select(country, year, sales) %>%
    mutate(type = 'cv')
df21 <- rbind(pc, cv) %>%
    filter(is.na(country)==F) %>%
    mutate(
        country = str_replace(country, '\\*', ''), 
        year = as.numeric(str_replace(year, "Q1-Q4 ", ""))) %>% 
    filter(year > 2019)

# Merge years
df <- rbind(df19, df21)

# Fix country name typos
countryNames <- data.frame(
    country = c(
        'Switzerland (+Fl)', 'Congo Kinshasa', 'Moldavia',
        'United States Of America', 'Azerbaidjan', 'Cambodge',
        'Hong-Kong', 'Irak', 'Kazakstan', 'Kirghizistan', 'Tadjikistan',
        'Tahiti', 'Tukmenistan', 'Burkina', 'Guiana (French)', 'Guiana',
        'Bulgaria1', 'Mexico2'),
    goodName = c(
        'Switzerland', 'Congo', 'Moldova', 'United States',
        'Azerbaijan', 'Cambodia', 'Hong Kong', 'Iraq', 'Kazakhstan',
        'Kyrgyzstan', 'Tajikistan', 'French Polynesia',
        'Turkmenistan', 'Burkina Faso', 'French Guiana', 'French Guiana',
        'Bulgaria', 'Mexico'))
df <- df %>%
    left_join(countryNames, by = "country") %>%
    mutate(
        goodName=ifelse(is.na(goodName), country, as.character(goodName)),
        country=goodName) %>%
    select(-goodName)

# Add region and continent data
worldRegions <- read_csv(file.path('regionsData', 'worldRegions.csv'))
df <- df %>%
    left_join(worldRegions, by = "country") %>%
    arrange(year, country)

# Separate out countries and regions
regions <- df %>%
    filter(is.na(region) == T) %>%
    mutate(region = country) %>%
    select(region, year, sales, type) %>% 
    mutate(
        region = ifelse(
            region %in% c(
                "Eu 27 Countries + Efta + Uk", "Eu 28 Countries + Efta"), 
            "EU", region
        )
    ) %>% 
    filter(! region %in% c('Eu 15 Countries + Efta', 'Europe New Members')) %>% 
    arrange(region, year) 
countries <- df %>%
    filter(is.na(region) == F) %>% 
    arrange(country)

# Save data frames
write_csv(countries, file.path('salesData', 'salesDataByCountry.csv'))
write_csv(regions, file.path('salesData', 'salesDataByRegion.csv'))







# Save the datasets
usethis::use_data(cars_us, overwrite = TRUE)
usethis::use_data(cars_china, overwrite = TRUE)


data-raw/format_producton.R
library(tidyverse)
library(rvest)
library(janitor)

parse_table <- function(year) {
    url_base <- "https://www.oica.net/category/production-statistics/"
    url <- paste0(url_base, year, "-statistics/")
    df <- read_html(url) %>%
        html_table(header = TRUE, trim = TRUE)
    df <- clean_names(df[[1]]) %>%
        mutate(year = year)
    return(df)
}

tables <- lapply(seq(2006, 2021), parse_table)

production <- do.call(rbind, tables) %>%
    mutate(
        country = str_to_title(country_region),
        pv = parse_number(cars),
        cv = parse_number(commercial_vehicles, na = c("N.A.", "-"))) %>%
    select(year, country, pv, cv) %>%
    filter(! country %in% c("Total", "Totals")) %>%
    pivot_longer(
        names_to = "type",
        values_to = "n",
        pv:cv
    ) %>%
    # Harmonize some country names
    mutate(
        country = case_when(
            country == "Czech Rep." ~ "Czech Republic",
            country == "Usa" ~ "USA",
            country == "Uk" ~ "United Kingdom",
            country == "Supplementary" ~ "Others",
            TRUE ~ country
        )
    )

# Save as csv
write_csv(production, file.path("data-raw", "production.csv"))

# Save the datasets
usethis::use_data(production, overwrite = TRUE)


data-raw/format_data.R
library(tidyverse)
library(rvest)
library(janitor)
library(readxl)
options(dplyr.width = Inf)

# Standard country / area codes from UN

world_regions <- read_excel(
    file.path('data-raw', 'raw', 'un_country_data.xlsx')) %>%
    clean_names() %>%
    select(
        country = country_or_area,
        region = region_name,
        subregion = sub_region_name,
        intermediate_region = intermediate_region_name,
        least_developed = least_developed_countries_ldc,
        land_locked_developing = land_locked_developing_countries_lldc,
        small_island_developing = small_island_developing_states_sids,
        code_region = region_code,
        code_subregion = sub_region_code,
        code_intermediate_region = intermediate_region_code,
        code_m49 = m49_code,
        code_iso_alpha2 = iso_alpha2_code,
        code_iso_alpha3 = iso_alpha3_code
    ) %>%
    mutate(
        least_developed = ifelse(is.na(least_developed), 0, 1),
        land_locked_developing = ifelse(is.na(land_locked_developing), 0, 1),
        small_island_developing = ifelse(is.na(small_island_developing), 0, 1)
    ) %>%
    # Make column for mis-matching country names to match OICA names
    mutate(
        country_oica = case_when(
            country == 'Bolivia (Plurinational State of)' ~ 'Bolivia',
            country == 'Bosnia and Herzegovina' ~ 'Bosnia',
            country == 'Brunei Darussalam' ~ 'Brunei',
            country == 'China, Hong Kong Special Administrative Region' ~ 'Hong Kong',
            country == 'Czechia' ~ 'Czech Republic',
            country == 'Iran (Islamic Republic of)' ~ 'Iran',
            country == "Lao People's Democratic Republic" ~ 'Laos',
            country == 'Republic of Korea' ~ 'South Korea',
            country == 'Republic of Moldova' ~ 'Moldova',
            country == 'Réunion' ~ 'Reunion',
            country == 'Russian Federation' ~ 'Russia',
            country == 'State of Palestine' ~ 'Palestine',
            country == 'Syrian Arab Republic' ~ 'Syria',
            country == 'United Kingdom of Great Britain and Northern Ireland' ~ 'United Kingdom',
            country == 'United Republic of Tanzania' ~ 'Tanzania',
            country == 'United States of America' ~ 'USA',
            country == 'Venezuela (Bolivarian Republic of)' ~ 'Venezuela',
            country == 'Viet Nam' ~ 'Vietnam',
            TRUE ~ country
        )
    ) %>%
    # Add Taiwan
    rbind(data.frame(
        country = 'Taiwan',
        region = 'Asia',
        subregion = 'Eastern Asia',
        intermediate_region = NA,
        least_developed = 0,
        land_locked_developing = 0,
        small_island_developing = 0,
        code_region = 142,
        code_subregion = 30,
        code_intermediate_region = NA,
        code_m49 = NA,
        code_iso_alpha2 = 'CN-TW',
        code_iso_alpha3 = 'TWN',
        country_oica = 'Taiwan'
    ))

write_csv(world_regions, file.path("data-raw", "world_regions.csv"))




# Production Data ----

# Function for scraping the production data table out of the web page
# for a given year
parse_table <- function(year) {
    url_base <- "https://www.oica.net/category/production-statistics/"
    url <- paste0(url_base, year, "-statistics/")
    df <- read_html(url) %>%
        html_table(header = TRUE, trim = TRUE)
    df <- clean_names(df[[1]]) %>%
        mutate(year = year)
    return(df)
}

# Get all the tables
tables <- lapply(seq(2006, 2021), parse_table)

# Make the data frame
production <- do.call(rbind, tables) %>%
    mutate(
        country = str_to_title(country_region),
        pv = parse_number(cars),
        cv = parse_number(commercial_vehicles, na = c("N.A.", "-"))) %>%
    select(year, country, pv, cv) %>%
    filter(! country %in% c("Total", "Totals")) %>%
    pivot_longer(
        names_to = "type",
        values_to = "n",
        pv:cv
    ) %>%
    # Harmonize some country names
    mutate(
        country = case_when(
            country == "Czech Rep." ~ "Czech Republic",
            country == "Usa" ~ "USA",
            country == "Uk" ~ "United Kingdom",
            country == "Supplementary" ~ "Others",
            TRUE ~ country
        )
    ) %>%
    # Join region data
    left_join(
        world_regions %>%
            select(-country) %>%
            rename(country = country_oica),
        by = 'country')

# Save as csv
write_csv(production, file.path("data-raw", "production.csv"))



# Sales Data ----

# Raw data downloaded from
# https://www.oica.net/category/sales-statistics/

# pc = passenger cars
# cv = commercial vehicles

# 2005 - 2019

pc <- read_excel(file.path('data-raw', 'raw', 'pc-sales-2019.xlsx'), skip = 5)
pc$country <- str_to_title(pc$'REGIONS/COUNTRIES')
pc <- pc %>%
    gather(year, sales, '2005':'2019') %>%
    select(country, year, sales) %>%
    mutate(type = 'pc')
cv <- read_excel(file.path('data-raw', 'raw', 'cv-sales-2019.xlsx'), skip = 5)
cv$country <- str_to_title(cv$'REGIONS/COUNTRIES')
cv <- cv %>%
    gather(year, sales, '2005':'2019') %>%
    select(country, year, sales) %>%
    mutate(type = 'cv')
df_05_19 <- rbind(pc, cv) %>%
    filter(is.na(country)==F) %>%
    mutate(
        country = str_replace(country, '\\*', ''),
        year = as.numeric(year)) %>%
    # Drop footnote rows
    filter(
        country != '1 Only Lv',
        country != '2 Including Heavy Trucks, Buses And Coaches'
    )

# 2020 - 2021
pc <- read_excel(file.path('data-raw', 'raw', 'pc-sales-2021.xlsx'), skip = 3)
pc$country <- str_to_title(pc$'REGIONS/COUNTRIES')
pc <- pc %>%
    gather(year, sales, 'Q1-Q4 2019':'Q1-Q4 2021') %>%
    select(country, year, sales) %>%
    mutate(type = 'pc')
cv <- read_excel(file.path('data-raw', 'raw', 'cv-sales-2021.xlsx'), skip = 3)
cv$country <- str_to_title(cv$'REGIONS/COUNTRIES')
cv <- cv %>%
    gather(year, sales, 'Q1-Q4 2019':'Q1-Q4 2021') %>%
    select(country, year, sales) %>%
    mutate(type = 'cv')
df_20_21 <- rbind(pc, cv) %>%
    filter(is.na(country)==F) %>%
    mutate(
        country = str_replace(country, '\\*', ''),
        year = as.numeric(str_replace(year, "Q1-Q4 ", ""))) %>%
    filter(year > 2019)

# Merge years and fix country name irregularities
df <- rbind(df_05_19, df_20_21) %>%
    mutate(
        country = case_when(
            country == 'Switzerland (+Fl)' ~ 'Switzerland',
            country == 'Congo Kinshasa' ~ 'Congo',
            country == 'Moldavia' ~ 'Moldova',
            country == 'United States Of America' ~ 'USA',
            country == 'Azerbaidjan' ~ 'Azerbaijan',
            country == 'Cambodge' ~ 'Cambodia',
            country == 'Hong-Kong' ~ 'Hong Kong',
            country == 'Irak' ~ 'Iraq',
            country == 'Kazakstan' ~ 'Kazakhstan',
            country == 'Kirghizistan' ~ 'Kyrgyzstan',
            country == 'Tadjikistan' ~ 'Tajikistan',
            country == 'Tahiti French Polynesia' ~ 'French Polynesia',
            country == 'Tahiti' ~ 'French Polynesia',
            country == 'Tukmenistan' ~ 'Turkmenistan',
            country == 'Trinidad' ~ 'Trinidad and Tobago',
            country == 'Burkina' ~ 'Burkina Faso',
            country == 'Guiana (French)' ~ 'French Guiana',
            country == 'Guiana' ~ 'French Guiana',
            country == 'Bulgaria1' ~ 'Bulgaria',
            country == 'Mexico2' ~ 'Mexico',
            TRUE ~ country
        )
    )

# Join region data
df <- df %>%
    left_join(
    world_regions %>%
        select(-country) %>%
        rename(country = country_oica),
    by = 'country') %>%
    select(year, country, type, n = sales, everything())

# Separate out countries and regions
sales_region <- df %>%
    filter(is.na(region) == TRUE) %>%
    select(year, region = country, type, n) %>%
    mutate(
        region = ifelse(
            region %in% c(
                "Eu 27 Countries + Efta + Uk", "Eu 28 Countries + Efta"),
            "EU", region
        )
    ) %>%
    filter(! region %in% c('Eu 15 Countries + Efta', 'Europe New Members')) %>%
    arrange(region, year)

sales_country <- df %>%
    filter(is.na(region) == F) %>%
    arrange(country)

# Save data frames
write_csv(sales_country, file.path('data-raw', 'sales_country.csv'))
write_csv(sales_region, file.path('data-raw', 'sales_region.csv'))



# Save the datasets
usethis::use_data(production, overwrite = TRUE)
usethis::use_data(sales_country, overwrite = TRUE)
usethis::use_data(sales_region, overwrite = TRUE)
